src/js/scripts/generate-features.js

```diff
@@ -0,0 +1,142 @@
+            .join("\n");
+    }
+
+    static validateTestCase(testCase, schema) {
+        // Convert the template schema directly to JSON Schema format
+        const jsonSchema = {
+            type: "object",
+            properties: schema,
+            required: Object.keys(schema),
+            additionalProperties: true,
+            $id: `testCase_${Math.random()}`,
+        };
+
+        const { isValid, errors } = validate(testCase, jsonSchema);
+
+        if (!isValid) {
+            const errorDetails = errors
+                .map((err) => `${err.instancePath} ${err.message}`)
+                .join("; ");
+            throw new Error(
+                `Validation failed for test case "${testCase.feature_name}": ${errorDetails}`,
+            );
+        }
+
+        return true;
+    }
+
+    static expandTemplate(template, context) {
+        return template.replace(/\${([^}]+)}/g, (match, key) => {
+            const trimmedKey = key.trim();
+
+            if (trimmedKey.endsWith("_table")) {
+                const arrayKey = trimmedKey.replace("_table", "");
+                const array = context[arrayKey];
+                if (array && Array.isArray(array)) {
+                    const columns = array[0] ? Object.keys(array[0]) : [];
+                    return TestFeatureGenerator.generateTable(array, columns);
+                }
+                return "";
+            }
+
+            return context[trimmedKey] !== undefined ? String(context[trimmedKey]) : match;
+        });
+    }
+
+    processTemplateFile(yamlPath) {
+        try {
+            const yamlContent = fs.readFileSync(yamlPath, "utf8");
+            const config = yaml.load(yamlContent);
+
+            const templatePath = path.join(this.templateDir, config.template);
+            const templateContent = fs.readFileSync(templatePath, "utf8");
+
+            config.cases.forEach((testCase) => {
+                TestFeatureGenerator.validateTestCase(testCase, config.templateSchema);
+
+                const context = {
+                    ...testCase,
+                    feature_path: config.feature_path,
+                };
+
+                const expandedContent = TestFeatureGenerator.expandTemplate(
+                    templateContent,
+                    context,
+                );
+                const featuresDir = path.join(this.outputDirRoot, config.feature_path);
+                const outputPath = path.join(featuresDir, `${testCase.feature_name}.feature`);
+
+                if (!fs.existsSync(featuresDir)) {
+                    fs.mkdirSync(featuresDir, { recursive: true });
+                }
+
+                fs.writeFileSync(outputPath, expandedContent);
+                console.log(`Generated: ${outputPath}`);
+            });
+        } catch (error) {
+            console.error(`Error processing ${yamlPath}:`, error);
+        }
+    }
+
+    generate() {
+        const yamlFiles = glob.sync(path.join(this.templateDir, "*.yaml"));
```

Comment:
We should have a class that deals with a single yaml file and then use it in a loop inside the script